torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2291,6 +2291,15 @@ options :class:`~torch.distributed.ProcessGroupNCCL.Options`).
           .def(
               "comm_split_count",
               &::c10d::ProcessGroupNCCL::getCommSplitCounter)
+          .def(
+              "_reset_nccl_collective_timeout",
+              [](const c10::intrusive_ptr<::c10d::ProcessGroupNCCL>& self,
+                 int timeout_mil_sec) {
```

Comment:
The signature of the function is open for discussion here.

Reply:
i'd suggest 'set' over 'reset'.  (reset usually means "put back to zero or put back to some baseline" while set requires a new value.

and maybe instead of `nccl_collective_timeout` just `default_timeout` (since 'nccl' is already assumed, this API only exists for nccl pg backend afaiu, and this timeout won't apply to all collectives just new ones).

i think its best if we expose a 'timedelta' obj on the python side for consistency with our other timeout apis, which lets users conveniently specify in sec or whatever they want.  But if it's not easy to pybind a timedelta obj, then maybe this is fine.

Reply:
agree with `set` and with pybind `timedelta` or floats should be automatically converted std::chrono::duration so no need to have the arg as `timeout_mil_sec` just `timeout` and the user can use either

Reply:
Actually passing `3000` in python does not work if I set `std::chrono::milliseconds timeout`